src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -159,6 +181,87 @@ private string GetRootPathWithGuid_NoLock()
             return new MetadataAsSourceFile(fileInfo.TemporaryFilePath, navigateLocation, documentName, documentTooltip);
         }
 
+        private async Task<Document> DecompileSymbolAsync(Document temporaryDocument, ISymbol symbol, CancellationToken cancellationToken)
+        {
+            // Get the name of the type the symbol is in:
+            var containingOrThis = symbol.GetContainingTypeOrThis();
+            var total = GetFullReflectionName(containingOrThis);
+
+            var model = await temporaryDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            // TODO : retrieve path to actual assembly instead of reference assembly:
+            var reference = model.Compilation.GetMetadataReference(symbol.ContainingAssembly);
```

Comment:
Should this be calling `MetadataAsSourceHelpers.GetAssemblyDisplay`? Or unify that somehow with the other code? There's an interesting comment in `MetadataAsSourceHelpers` pointing to #6859 asking why it's not just calling GetMetadataReference. I worry that either if there is some subtle reason the shenanigans in the helpers is necessary, then you'll want that same thing here. (And if it's unnecessary, congrats, you've confirmed you can delete some code.)

Reply:
➡️ Filed #23478

Reply:
Why a bug to track? This is potentially a bug being introduced. We should resolve prior to merge.